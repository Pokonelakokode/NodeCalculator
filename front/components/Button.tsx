import * as React from "react";

interface IProps {
    label: string;
    col: number;
    onClick: any;
}

const Button: React.FC<IProps> = ({label, col, onClick}) => {
    return (
        <div className={`col-${col} mt-2`}>
            <div className="numeric" onClick={onClick}>
                {label}
            </div>
        </div>
    );
};

export default Button;
