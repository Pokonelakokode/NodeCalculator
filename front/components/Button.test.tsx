import React from "react";
import {create} from "react-test-renderer";
import Button from "./Button";
import { mount } from "enzyme";

describe("Button", () => {
    it("should be able to render", () => {
        const mockFn = jest.fn();
        const component = create(
            <Button col={2} label={"TEST"} onClick={mockFn}/>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("should run a function when clicked", () => {
        const fn = jest.fn();
        const wrapper = mount(
            <Button onClick={fn} label="TEST" col={2}/>,
        );
        wrapper
            .find(".numeric")
            .simulate("click");
        expect(fn).toBeCalled();
    });
});
