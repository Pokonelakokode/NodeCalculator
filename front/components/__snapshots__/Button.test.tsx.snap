// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be able to render 1`] = `
<div
  className="col-2 mt-2"
>
  <div
    className="numeric"
    onClick={[MockFunction]}
  >
    TEST
  </div>
</div>
`;
